physical/raft/raft.go

```diff
@@ -474,9 +480,55 @@ func (b *RaftBackend) Close() error {
 		return err
 	}
 
+	close(b.statsDoneCh)
+
 	return nil
 }
 
+func (b *RaftBackend) collectMetrics() {
+	if b.metricSink == nil {
+		b.logger.Warn("collectMetrics() called without a metricSink set, aborting")
+		return
+	}
+
+	b.l.Lock()
+	b.statsDoneCh = make(chan struct{})
+	b.l.Unlock()
+
+	go func() {
+		for {
+			select {
+			case <-b.statsDoneCh:
+				return
+			default:
```

Comment:
Instead of a sleep, why not use a ticker?  That way we won't wait up to a second (or whatever the interval is) to notice we've been cancelled.

Reply:
Sure